### DISCUSSION TITLE
If i want replace openjdk with ikvm at idea, how should i do it.

### DISCUSSION THREAD

**Discussion by base-linsyd** [ikvmnet/ikvm#316] [Q&A]
Maybe this description is wrong .  my english is poor.

If i selected the ikvm file folder ,  the idea tell me  

![image](https://user-images.githubusercontent.com/63432702/235087919-b0b6d141-4846-4444-ab25-095b05859aab.png)

**Comment by wasabii** [comment]
What application is that? I'd imagine it depends on whether that application lets you do that.

**Reply by base-linsyd** [reply]
IntelliJ IDEA,  it may have verified the folder structure. If I copy the jre directory over, I can select it normally, but it cannot be used normally.

**Reply by wasabii** [reply]
Hmm. Yeah. Dunno what it's looking for.

**Reply by base-linsyd** [reply]
If I copy the openjdk jre directory over on ikvm sub directory, it will be selected normally, but cant work normally. Are there some jre functions that have not been implemented in ikvm ? for example , javaw.exe.

**Reply by wasabii** [reply]
You have to know what intellij is looking for.

**Reply by base-linsyd** [reply]
ok i will try it.

**Reply by wasabii** [reply]
So looking over this I see two possibilities.

First, we don't have a separate JRE directory. Maybe IntelliJ cares. The convention of having a separate JRE directory doesn't really matter for IKVM, so I didn't do it. And also, that convention was removed in JDK9+.

Second, we are missing some files IntelliJ might be looking for: tools.jar, rt.jar, etc. Because we don't package JARs. Our core libraries are in .NET assemblies in bin/. Maybe it checks that these files exist.

Making the JRE directory yourself should be possible: just copy the executables from the JDK twice. Kinda silly.

We can't help with tools.jar or rt.jar, since we simply don't have that. It wouldn't surprise me too much if it wanted rt.jar. But we can't do anything about that.

That said, we have all the executables expected for a JDK. `javaw` is a Windows optional version of java.exe. We don't need it.

Best you can do is consult with the IntelliJ developers to figure out if you can make it work.

**Reply by wasabii** [reply]
Stepping back though, I gotta wonder why you're bothering to try to use IKVM as the compiler for IntelliJ though.

**Reply by base-linsyd** [reply]
I wanted to test if ikvm could be used as the compiler for IntelliJ because I saw someone running Minecraft using ikvm.

**Reply by wasabii** [reply]
Well those seem like different things. Many people in the past have tried Minecraft. Different people have had different issues. #208

**Reply by base-linsyd** [reply]
ok. i will take a look☺️☺️☺️
